Jenkinsfile

```diff
@@ -0,0 +1,347 @@
+#!/usr/bin/env groovy
+
+// These are the platforms we are building against.
+def platforms = ["linux", "mac", "windows"]
+
+// This generates the launchConfig for a particular build.
+def generateConfig(deploymentName, templateUrl) {
+    return """
+---
+launch_config_version: 1
+deployment_name: ${deploymentName}
+template_url: ${templateUrl}
+provider: aws
+aws_region: us-west-1
+template_parameters:
+    KeyName: default
+    AdminLocation: 0.0.0.0/0
+    PublicSlaveInstanceCount: 1
+    SlaveInstanceCount: 1
+"""
+}
+
+
+// This class abstracts away the functions required to create a test cluster
+// for each of our platforms.
+class TestCluster implements Serializable {
+    WorkflowScript script
+    String platform
+    int createAttempts
+
+    TestCluster(WorkflowScript script, String platform) {
+        this.script = script
+        this.platform = platform
+        this.createAttempts = 0
+    }
+
+    def _create() {
```

Comment:
i dont understand the naming patterns here. Why are we duplicating names and using underscores inconsistently?